src/app/clusters/soil-measurement-server/soil-measurement-server.h

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/data-model/List.h>
+#include <cstdint>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace SoilMeasurement {
+
+inline constexpr uint16_t kClusterRevision = 1;
```

Comment:
This should come from the relevant Metadata.h header to pick up what's in the XML.

Reply:
Ok, I followed the pattern used by the others clusters here. Let me see if I can figure out how to do this.

Reply:
The other clusters could not do Metadata.h, because it's new.  Do what https://github.com/project-chip/connectedhomeip/pull/38726/files is doing: include `<clusters/SoilMeasurement/Metadata.h>` and then `kRevision` in the `chip::app::Clusters::SoilMeasurement` namespace is the revision in the XML.

Reply:
Understood, thanks for the guidance.

Reply:
Sorted in latest commit, resolving this comment for now.